ekncontent/ekncontent/eknc-utils.c

```diff
@@ -14,6 +16,86 @@
  * Utility functions for the Endless Knowledge Content library.
  */
 
+/**
+ * eknc_utils_append_gparam_from_json_node:
+ *
+ * For use instantiating a GObject from json data. Marshals the contents
+ * of a JsonNode into a GParameter matching the GParamSpec pspec. Appends
+ * that GParameter to a GArray, for eventual use with g_object_newv.
+ */
+void
+eknc_utils_append_gparam_from_json_node (JsonNode *node,
+                                         GParamSpec *pspec,
+                                         GArray *params)
+{
+  g_autoptr(GError) error = NULL;
+  GParameter param = { NULL, { 0, }};
+
+  if (!node)
+    return;
+
+  param.name = pspec->name;
+  g_value_init (&param.value, pspec->value_type);
+
+  if (G_IS_PARAM_SPEC_VARIANT (pspec))
+    {
+      GVariantType *type = G_PARAM_SPEC_VARIANT (pspec)->type;
+      GVariant *variant = json_gvariant_deserialize (node,
+                                                     g_variant_type_peek_string (type),
+                                                     &error);
+      g_value_set_variant (&param.value, variant);
+    }
+  else if (JSON_NODE_HOLDS_VALUE (node))
+    {
+      GValue value = { 0, };
+      GType type = json_node_get_value_type (node);
+
+      // For some reason some of our integer properties are stored in json as
+      // strings. Try to parse if we hit this case.
```

Comment:
Just an inconsistency in our data. A lot of our other number (duration, width, height) are store in our json metadata as string. The toc item `hasIndex` is the only thing I can find that's an actual number there.

Would be great to cleanup on the server side either all number in json as number or none. Then we'd probably need to keep a compat layer, but don't want this issue to extend to changing our on disk data format.

Can open up some follow ups though.